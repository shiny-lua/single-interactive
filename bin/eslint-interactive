#!/usr/bin/env node -r source-map-support/register

require('../dist');
