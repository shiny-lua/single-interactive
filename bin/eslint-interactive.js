#!/usr/bin/env -S node --enable-source-maps --unhandled-rejections=strict --experimental-import-meta-resolve

import { run } from '../dist/index.js';

run({
  argv: process.argv,
}).catch((error) => {
  console.error(error);
  process.exit(1);
});
