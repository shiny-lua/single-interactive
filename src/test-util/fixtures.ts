import { exec } from 'child_process';
import { dirname, join } from 'path';
import { fileURLToPath } from 'url';
import { promisify } from 'util';

const execPromise = promisify(exec);

const cwd = join(dirname(fileURLToPath(import.meta.url)), '..', '..');

/**
 * Returns a string containing the stitched together contents of the file modified by fix.
 * To make the snapshot easier to read, the name of the file is inserted at the beginning of the contents of each file.
 */
export async function getSnapshotOfChangedFiles(): Promise<string> {
  const { stdout } = await execPromise(`diff -qr fixtures fixtures-tmp | cut -d " " -f 4 | xargs tail -n +1`, {
    cwd,
  });
  return stdout.toString();
}

export async function setupFixturesCopy() {
  await execPromise(`rm -rf fixtures-tmp && cp -r fixtures fixtures-tmp`, { cwd });
}

export async function cleanupFixturesCopy() {
  await execPromise(`rm -rf fixtures-tmp`, { cwd });
}
