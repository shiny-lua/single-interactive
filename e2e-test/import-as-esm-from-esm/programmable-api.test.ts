import { dirname, join } from 'path';
import { fileURLToPath } from 'url';
import { Core, takeRuleStatistics } from 'eslint-interactive';
import stripAnsi from 'strip-ansi';
import { cleanupFixturesCopy, getSnapshotOfChangedFiles, setupFixturesCopy } from '../../src/test-util/fixtures.js';

beforeEach(async () => {
  await setupFixturesCopy();
});

afterEach(async () => {
  await cleanupFixturesCopy();
});

test('Programmable API', async () => {
  const core = new Core({
    patterns: ['fixtures-tmp'],
    cwd: join(dirname(fileURLToPath(import.meta.url)), '..', '..'),
  });
  const results = await core.lint();

  expect(stripAnsi(core.formatResultSummary(results))).toMatchSnapshot();
  const statistics = takeRuleStatistics(results);
  expect(statistics).toMatchSnapshot();

  await core.applyAutoFixes(results, ['semi']);
  expect(await getSnapshotOfChangedFiles()).toMatchSnapshot();
});
