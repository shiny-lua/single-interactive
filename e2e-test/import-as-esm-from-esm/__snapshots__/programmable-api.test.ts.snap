// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Programmable API 1`] = `
"- 7 files (0 file passed, 7 files failed) checked.
- 44 problems (43 errors, 1 warning) found.
┌─────────────────────────────┬───────┬─────────┬────────────┬─────────────────┐
│ Rule                        │ Error │ Warning │ is fixable │ has suggestions │
├─────────────────────────────┼───────┼─────────┼────────────┼─────────────────┤
│ ban-exponentiation-operator │ 7     │ 0       │ 0          │ 0               │
├─────────────────────────────┼───────┼─────────┼────────────┼─────────────────┤
│ semi                        │ 12    │ 1       │ 13         │ 0               │
├─────────────────────────────┼───────┼─────────┼────────────┼─────────────────┤
│ import/order                │ 3     │ 0       │ 3          │ 0               │
├─────────────────────────────┼───────┼─────────┼────────────┼─────────────────┤
│ prefer-const                │ 4     │ 0       │ 4          │ 0               │
├─────────────────────────────┼───────┼─────────┼────────────┼─────────────────┤
│ no-unused-vars              │ 7     │ 0       │ 0          │ 0               │
├─────────────────────────────┼───────┼─────────┼────────────┼─────────────────┤
│ no-useless-escape           │ 5     │ 0       │ 0          │ 5               │
├─────────────────────────────┼───────┼─────────┼────────────┼─────────────────┤
│ no-unsafe-negation          │ 5     │ 0       │ 0          │ 5               │
└─────────────────────────────┴───────┴─────────┴────────────┴─────────────────┘"
`;

exports[`Programmable API 2`] = `
Array [
  Object {
    "errorCount": 7,
    "hasSuggestionsCount": 0,
    "hasSuggestionsErrorCount": 0,
    "hasSuggestionsWarningCount": 0,
    "isFixableCount": 0,
    "isFixableErrorCount": 0,
    "isFixableWarningCount": 0,
    "ruleId": "ban-exponentiation-operator",
    "warningCount": 0,
  },
  Object {
    "errorCount": 12,
    "hasSuggestionsCount": 0,
    "hasSuggestionsErrorCount": 0,
    "hasSuggestionsWarningCount": 0,
    "isFixableCount": 13,
    "isFixableErrorCount": 12,
    "isFixableWarningCount": 1,
    "ruleId": "semi",
    "warningCount": 1,
  },
  Object {
    "errorCount": 3,
    "hasSuggestionsCount": 0,
    "hasSuggestionsErrorCount": 0,
    "hasSuggestionsWarningCount": 0,
    "isFixableCount": 3,
    "isFixableErrorCount": 3,
    "isFixableWarningCount": 0,
    "ruleId": "import/order",
    "warningCount": 0,
  },
  Object {
    "errorCount": 4,
    "hasSuggestionsCount": 0,
    "hasSuggestionsErrorCount": 0,
    "hasSuggestionsWarningCount": 0,
    "isFixableCount": 4,
    "isFixableErrorCount": 4,
    "isFixableWarningCount": 0,
    "ruleId": "prefer-const",
    "warningCount": 0,
  },
  Object {
    "errorCount": 7,
    "hasSuggestionsCount": 0,
    "hasSuggestionsErrorCount": 0,
    "hasSuggestionsWarningCount": 0,
    "isFixableCount": 0,
    "isFixableErrorCount": 0,
    "isFixableWarningCount": 0,
    "ruleId": "no-unused-vars",
    "warningCount": 0,
  },
  Object {
    "errorCount": 5,
    "hasSuggestionsCount": 5,
    "hasSuggestionsErrorCount": 5,
    "hasSuggestionsWarningCount": 0,
    "isFixableCount": 0,
    "isFixableErrorCount": 0,
    "isFixableWarningCount": 0,
    "ruleId": "no-useless-escape",
    "warningCount": 0,
  },
  Object {
    "errorCount": 5,
    "hasSuggestionsCount": 5,
    "hasSuggestionsErrorCount": 5,
    "hasSuggestionsWarningCount": 0,
    "isFixableCount": 0,
    "isFixableErrorCount": 0,
    "isFixableWarningCount": 0,
    "ruleId": "no-unsafe-negation",
    "warningCount": 0,
  },
]
`;

exports[`Programmable API 3`] = `
"==> fixtures-tmp/index.js <==
import { strictEqual } from 'assert';
import { add } from './lib/add';
import { exp } from './lib/exp';
import { mul } from './lib/mul';
import { sub } from './lib/sub';

const numA=add(1, 2);
const numB=add(3, 4);
const numC=sub(5, 6);
const numD=exp(2, 2);
const numE=mul(numA, numB, numC, numD);

strictEqual(numD, -84);

==> fixtures-tmp/lib/add.js <==
/* eslint semi: 1 */

export function add(a, b) {
  return a + b;
}

==> fixtures-tmp/lib/shebang.js <==
#!/usr/bin/env node

console.log();
"
`;
